libs/pageserver_api/src/key.rs

```diff
@@ -103,7 +122,7 @@ impl Key {
     /// we can assume that only some predefined namespace OIDs are used which can fit in u16
     pub fn to_i128(&self) -> i128 {
         assert!(self.field2 < 0xFFFF || self.field2 == 0xFFFFFFFF || self.field2 == 0x22222222);
-        (((self.field1 & 0xf) as i128) << 120)
+        (((self.field1 & 0x7F) as i128) << 120)
```

Comment:
shouldn't this be 0xff to make it bijective?

Reply:
It's i128 and there are many places in the code to create i128 ranges. If we map to i128 negative number, there will be some panics. In the future we should refactor this into u128, but it would be a large pull request and I don't plan to do it in this week.